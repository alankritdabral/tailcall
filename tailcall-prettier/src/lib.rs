
use std::io::Write;
use std::process::{Command, Stdio};

use anyhow::{anyhow, Result};

pub fn format_with_prettier<T: AsRef<str>>(code: T, file_ty: T) -> Result<String> {
    let mut child = Command::new("prettier")
        .arg("--stdin-filepath")
        .arg(file_ty.as_ref())
        .stdin(Stdio::piped())
        .stdout(Stdio::piped())
        .spawn()?;

    if let Some(ref mut stdin) = child.stdin {
        stdin.write_all(code.as_ref().as_bytes())?;
    }

    let output = child.wait_with_output()?;
    if output.status.success() {
        Ok(String::from_utf8(output.stdout)?)
    } else {
        Err(anyhow!("Prettier formatting failed"))
    }
}

#[cfg(test)]
mod tests {
    use crate::format_with_prettier;

    #[test]
    fn test_js() -> anyhow::Result<()> {
        let prettier = format_with_prettier("const x={a:3};", "file.ts")?;
        insta::assert_snapshot!(prettier);
        Ok(())
    }
}
