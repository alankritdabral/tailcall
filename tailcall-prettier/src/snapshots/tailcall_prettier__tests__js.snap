---
source: tailcall-prettier/src/lib.rs
expression: prettier
---
const x = {a: 3}
